homeassistant/components/microbot_push/config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Adds config flow for MicroBot."""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Any
+
+from bleak.backends.device import BLEDevice
+from microbot import (  # pylint: disable=import-error
+    MicroBotAdvertisement,
+    MicroBotApiClient,
+    parse_advertisement_data,
+)
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_BDADDR, CONF_NAME, DEFAULT_RETRY_COUNT, DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+def format_unique_id(address: str) -> str:
+    """Format the unique ID for a switchbot."""
+    return address.replace(":", "").lower()
+
```

Comment:
This was for legacy compat with older BLE libs.  We can use the address as-is without this